### ISSUE TITLE
Cannot use streamed responses with blocking tools

### ISSUE THREAD

**Issue by cescoffier** [quarkiverse/quarkus-langchain4j#900]
Let's imagine an AI Service returning a `Multi<String>`. If the generation of the response uses tools, these tools cannot be blocking (basically, cannot use Panache or Hibernate). 

I tried:

- @RunOnVirtualThread (the _magic bullet(tm)_)
- @Blocking 
- `.runSubscriptionOn(...)`

None of them works. 

I think we need to find a proper way to handle that use case.

**Comment by geoand** [comment]
Right, this is similar to https://github.com/quarkiverse/quarkus-langchain4j/issues/837
